paddle/fluid/framework/new_executor/instruction/cond_instruction.cc

```diff
@@ -99,161 +85,128 @@ void GetOutsideOpInputs(
+            value_exec_info.HasValue(value),
+            true,
             phi::errors::PreconditionNotMet(
                 "input should in name map, [%d] 'th input of [%s] op",
                 i,
-                "if op"));
-        std::vector<int> inputs_id = GetValueIds(value,
-                                                 inner_scope,
-                                                 value_2_var_name,
-                                                 var_name_2_id,
-                                                 variable_2_var_name);
+                op->name()));
+        std::vector<int> inputs_id = GetValueIds(value, value_exec_info);
 
         input_ids->emplace(value, inputs_id);
       }
     }
   }
 }
 
-CondInstruction::CondInstruction(
-    size_t id,
-    const platform::Place& place,
-    pir::Operation* op,
-    Scope* scope,
-    Scope* local_scope,
-    ValueExecutionInfo* parent_exe_info,
-    const std::map<pir::Block*, paddle::framework::Scope*>& sub_blocks)
+CondInstruction::CondInstruction(size_t id,
+                                 const platform::Place& place,
+                                 pir::Operation* op,
+                                 ValueExecutionInfo* value_exec_info)
     : InstructionBase(id, place) {
-  op_ = op;
-  VLOG(6) << "finish process dist attributes";
-
-  SetKernelType(AnalyseOpFuncType(op, place));
-  VLOG(6) << "finish process analyse kernel type";
-
-  Scope* inner_scope = local_scope == nullptr ? scope : local_scope;
-
-  VLOG(6) << "finish process inputs outputs index";
-
   PADDLE_ENFORCE(
       op->isa<paddle::dialect::IfOp>(),
       phi::errors::PreconditionNotMet("Cond instruction only support if op"));
-
   auto if_op = op->dyn_cast<paddle::dialect::IfOp>();
+  op_ = op;
 
-  for (size_t i = 0; i < if_op.num_results(); ++i) {
-    if_op_outputs_.push_back(inner_scope->GetVar(
-        parent_exe_info->GetValue2VarName().at(if_op.result(i))));
-  }
+  SetKernelType(AnalyseOpFuncType(op, place));
+  VLOG(6) << "finish process analyse kernel type";
 
   auto cond_value = if_op.operand_source(0);
-  auto var_name = parent_exe_info->GetValue2VarName().at(cond_value);
-  cond_var = inner_scope->FindVar(var_name);
+  cond_var_ = value_exec_info->GetScope()->FindVar(
+      value_exec_info->GetValue2VarName().at(cond_value));
+  for (size_t i = 0; i < if_op.num_results(); ++i) {
+    output_vars_.push_back(value_exec_info->GetScope()->GetVar(
+        value_exec_info->GetValue2VarName().at(if_op.result(i))));
+  }
+  VLOG(6) << "finish process cond_var and output_vars";
 
   auto true_branch_block = if_op.true_block();
-  auto false_branch_block = if_op.false_block();
-
   auto true_branch_yied_inputs = GetYiedOpInputs(true_branch_block);
-  auto false_branch_yied_inputs = GetYiedOpInputs(false_branch_block);
-
-  auto true_scope = sub_blocks.at(true_branch_block);
-  true_branch_inter =
+  Scope* true_scope = &(value_exec_info->GetScope()->NewScope());
+  true_branch_inter_ =
```

Comment:
在哪里 delete 这个new出来的对象的？

Reply:
子 Scope 的生命周期由父Scope 管理，NewScope 的返回值是一个引用，该函数的定义是：
```c++
  /// Create a sub-scope. Returns a reference other than a pointer so
  /// to prevent from manual deletion.
  /// Mark it to const because that new kid scope cannot change parent scope.
  Scope& NewScope() const;
```

Reply:
我指的是 true_branch_inter_ 这个